### PR TITLE
[Doc Fixit 2020] Create frontend user guide

### PR DESCRIPTION
**Pull request by CaseyHillers**
Imported from flutter/flutter to be closer to the code. I'll send a follow up PR there to link here.

Added some common questions about the frontend I have seen over the past month.

# Issues

https://github.com/flutter/flutter/issues/68336

### CODE REVIEW

File: app_flutter/USER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+This dashboard loads a lot of data, and can take ~1 minute to get a response
+from the backend.
+
+### How do I update the baseline for a benchmark?
+
+1. Ensure that you are signed in with an AllowListedAccount with Cocoon
+2. Hover over top right corner of the benchmark graph of interest
+3. Click the magnifying glass
+4. A UI at the top will come down with a form to change baseline values
```
Comment (keyonghan) [inline app_flutter/USER_GUIDE.md:83]: shall these two be moved to `app/.`?

File: app_flutter/USER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+Tasks with exclamation points indicate a task that has been run multiple times.
+This indicates extra capacity used for this task. In some cases, this is an
+infra issues. If it's common for a task column to be filled with green tasks
+with exclamation marks, it indicates that task is flaky.
+
+### How do I view results for a release branch?
+
+Click the settings cog in the top right, and switch the branch via the dropdown.
+
+If the branch is not in the list, it has not propagated to Cocoon's backend.
```
Comment (keyonghan) [inline app_flutter/USER_GUIDE.md:59]: Do you want to add doc about the cache refresh?

File: app_flutter/USER_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Build dashboard
+
+The build page is accessible at https://flutter-dashboard.appspot.com/#/build.
+This page reports the build statuses of commits to the flutter/flutter repo.
+
+### Tree Closures
+
+The top navigation bar indicates the current tree status. If the tree is closed,
+a test has failed against the top of tree and flutter/flutter is currently not
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply app_flutter/USER_GUIDE.md]: Moved to `If the tree is closed, at least one test has failed against the top of tree...`

File: app_flutter/USER_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+### Why is a task stuck on "new task" status?
+
+The dashboard aggregates build results from multiple build environments,
+including Cirrus, LUCI, and DeviceLab. While DeviceLab attempts to
+tests every commit that goes into the `master` branch, other environments
+may skip some commits. LUCI may skip commits when they come in too fast.
+
+Flutter infra may skip running tests against a commit if there are passing
+results from a more recent commit.
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply app_flutter/USER_GUIDE.md]: Good point, done.

File: app_flutter/USER_GUIDE.md (hunk 1 above)
Comment (CaseyHillers) [reply app_flutter/USER_GUIDE.md:87]: I opted for here as it's easier to link to one frontend user guide versus two different ones.

File: app_flutter/USER_GUIDE.md (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/USER_GUIDE.md:61]: Added link to the documentation in app_dart README

File: app_flutter/USER_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+to be added as an AllowListedAccount to Cocoon.
+
+When the DeviceLab migrates to LUCI, logs will be available without sign in.
+
+### Why is a task stuck on "new task" status?
+
+The dashboard aggregates build results from multiple build environments,
+including Cirrus, LUCI, and DeviceLab. While DeviceLab attempts to
+tests every commit that goes into the `master` branch, other environments
+may skip some commits. LUCI may skip commits when they come in too fast.
```
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply app_flutter/USER_GUIDE.md]: I'm happy to link to some backend documentation for those that are curious :-)

I don't think frontend users should be required to understand stages and how this data is sourced.

**Review by keyonghan — APPROVED** [review]
LGTM. Thanks for the updates.

**Pull request merged**